pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -223,6 +267,37 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
 		}
 		isvc.Status.PropagateStatus(v1beta1.PredictorComponent, status)
 	}
-
+	statusSpec, _ := isvc.Status.Components[v1beta1.PredictorComponent]
+	if rawDeployment {
+		podLabelValue = "isvc." + isvc.ObjectMeta.Name + "-predictor-default"
+	} else {
+		podLabelValue = statusSpec.LatestCreatedRevision
+	}
+	podList, err := p.listPodsByLabel(isvc.ObjectMeta.Namespace, podLabelKey, podLabelValue)
+	if err != nil {
+		return errors.Wrapf(err, "fails to list inference service pods by label")
+	}
+	if len(podList.Items) > 0 {
+		isvc.Status.PropagateModelStatus(statusSpec, podList, rawDeployment)
+	}
 	return nil
 }
+
+func (p *Predictor) listPodsByLabel(namespace string, labelKey string, labelVal string) (*v1.PodList, error) {
```

Comment:
Can move this function to the utils file

Reply:
@yuzisun Updated, Thanks!